### ISSUE TITLE
Doesnt work in Electron apps that have used .asar to package themselves

### ISSUE THREAD

**Issue by pfrazee** [ipfs/js-ipfsd-ctl#94]
See https://github.com/electron/electron/issues/2708

The startDaemon() call never returns error or success. I'm fairly sure the issue is as in 2708: the relative paths fail.

**Comment by pfrazee** [comment]
I solved this by not using .asar. I havent figured out why Electron folks bother with the .asar, so I'm not sure what the downside is for my change.

**Comment by jbenet** [comment]
(My comment is Off topic)

> bother with .asar

I think it's supposed to be a bit better than tar, less error prone (stream
cutoff errors) and packed, etc. similar to our desires for .car

**Comment by pfrazee** [comment]
Yeah but there's no compression in .asar, and osx is already bundled into a
dmg. Maybe it's for windows and linux? not sure

**Comment by pfrazee** [comment]
Aha:

> To mitigate issues around long path names on Windows, slightly speed up require and conceal your source code from cursory inspection, you can choose to package your app into an asar archive with little changes to your source code.

http://electron.atom.io/docs/tutorial/application-packaging/

**Comment by pfrazee** [comment]
It turns out, they'd even documented this specific issue: http://electron.atom.io/docs/tutorial/application-packaging/#executing-binaries-inside-asar-archive

**Comment by diasdavid** [comment]
@haadcode do you run into this issue with orbit? How did you solve it?

**Comment by haadcode** [comment]
@diasdavid fixed it in the same way as @pfrazee did: not using asar.

**Comment by thisconnect** [comment]
It seems that the solution is to exclude go-ipfs-dep from node_modules during packaging. I got it working with `electron-packager`, see https://github.com/ipfs/js-ipfsd-ctl/pull/155

```
electron-packager ./ --overwrite --out ./dist --asar.unpackDir=node_modules/go-ipfs-dep/**
```

**Comment by diasdavid** [comment]
@thisconnect thanks for working on a interim solution and documenting it on the README 👍  https://github.com/ipfs/js-ipfsd-ctl#packaging

**Comment by thisconnect** [comment]
Can this issue be closed?
